src/python/pants/backend/python/providers/pyenv/target_types.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.engine.target import Field, Target
+
+
+class PyenvInstallSentinelField(Field):
+    none_is_valid_value = True
+    alias = "_sentinel"
+    help = "<internal>"
+    default = False
+
+
+class PyenvInstall(Target):
+    alias = "_pyenv_install"
+    help = "<internal target>"
+    core_fields = (PyenvInstallSentinelField,)
```

Comment:
this is a bit boiler-platey. not suggesting to do anything about it here and now, but if we find this pattern useful, it could make sense to streamline it a bit :)